src/services/ldap-config/service.js

```diff
@@ -0,0 +1,288 @@
+	 * access is denied.
+	 * To access the system data, the authentication information are proxied to the
+	 * System service, so that access is only granted to administrators and only to
+	 * systems of the own school.
+	 * @param {ObjectId} systemId ID of an existing system with LDAP config
+	 * @param {Object} params Feathers request params
+	 * @returns {Object} ldapConfig attribute of the given system
+	 */
+	async get(systemId, params) {
+		// Proxying the request params here to have the system service hooks check
+		// the permissions, filtering sensitive data, etc.
+		const system = await this.app.service('systems').get(systemId, params);
+		if (!system.ldapConfig || system.ldapConfig.provider !== 'general') {
+			throw new Forbidden('The requested system is not configurable with this endpoint');
+		}
+		return system.ldapConfig;
+	}
+
+	/**
+	 * `POST /ldap-config`
+	 * Creates a new system with the given LDAP config and attaches it to the
+	 * caller's school, if the config is validated successfully. The validation
+	 * result is returned.
+	 * If `validateOnly === true`, changes are not persisted to the systems and
+	 * school collection. Otherwise, a new system is created.
+	 * After creating the system, it is added as a login system to the caller's
+	 * school and (if `activate !== false`) the school's ldapSchoolIdentifier
+	 * attribute is set to the config's base path.
+	 * @param {Object} config an LDAP config (given as JSON in request body)
+	 * @param {Object} params Feathers request params
+	 * @returns {Object} validation result object
+	 * `{ ok: Boolean, errors: [], users: {}, classes: {} }`
+	 */
+	async create(config, params) {
+		return this.verifyAndSaveLdapConfig(config, this.getOptions(params));
+	}
+
+	/**
+	 * `PATCH /ldap-config/:systemId`
+	 * Patches an existing system with the given LDAP config and attaches it to
+	 * the caller's school if the config is validated successfully.
+	 * If `validateOnly === true`, changes are not persisted to the systems and
+	 * school collection. Otherwise, the system is patched.
+	 * After patching the system, if `activate !== false`, it is added as a login
+	 * system to the caller's school and the school's ldapSchoolIdentifier attribute
+	 * is set to the config's base path.
+	 * @param {ObjectId} id systemId of system to patch
+	 * @param {Object} config LDAP config object
+	 * @param {Object} params Feathers request params
+	 * @returns validation result object
+	 * `{ ok: Boolean, errors: [], users: {}, classes: {} }`
+	 */
+	async patch(id, config, params) {
+		this.systemId = id;
+		return this.verifyAndSaveLdapConfig(config, this.getOptions(params));
+	}
+
+	/**
+	 * Retrieves and returns options object to be used for #verifyAndSaveLdapConfig
+	 * @param {Object} params feathers request params
+	 * @returns {Object} options `{ school: School, activateSystem: Boolean, saveSystem: Boolean}`
+	 */
+	getOptions(params) {
```

Comment:
it is better to path the concrete params here so the caller can see which params should be used. `Params` parameter is too abstract :)